Client/Clique/Modules/EventDetail/EventDetailViewController.swift

```diff
@@ -20,32 +22,172 @@ class EventDetailViewController: UIViewController {
     @IBOutlet private var _eventParticipantsNum: UILabel!
     @IBOutlet private var _eventCost: UILabel!
     @IBOutlet private var _eventCategory: UILabel!
+    @IBOutlet private var buttonJoinEvent: UIButton!
+    @IBOutlet private var cosmosView: CosmosView!
+    
+    
+    private var eventServices = EventServices()
+    private var status = 0
+    private var alertMessage = ""
+    private var eventStatus = Event.status.done
     
     override func viewDidLoad() {
         super.viewDidLoad()
         _setupUI()
     }
+    
+    static func initializeDetailsViewController(with event: Event) -> UIViewController? {
+        let storyboard = UIStoryboard(name: String(describing: "EventDetail"), bundle: nil)
+        let viewController = storyboard.instantiateInitialViewController() as? EventDetailViewController
+        viewController?.event = event
+        return viewController
+    }
 }
 
 private extension EventDetailViewController {
     
     func _setupUI() {
-        guard let _event = self.event else { return }
-        
-        
-        _eventName.text = _event.name
-        _eventDescription.text = _event.description
-        _eventCreator.text = _event.creator.name + " " + _event.creator.surname
-        _eventLocation.text = _event.location
-        _eventTimestamp.text = _event.timestamp
-        _eventParticipantsNum.text = String(_event.participantNumber)
-        let currency = _event.currency
-        if let price = _event.cost {
+        _eventName.text = event.name
+        _eventDescription.text = event.description
+        _eventCreator.text = event.creator.name + " " + event.creator.surname
+        _eventLocation.text = event.location
+        _eventTimestamp.text = event.timeStampToString(timestamp: event.timestamp)
+        _eventParticipantsNum.text = String(event.participantNumber)
+        let currency = event.currency
+        if let price = event.cost {
             _eventCost.text = price == 0 ? Constants.Labels.labelFree : String(price) + " " + (currency ?? "")
         } else {
             _eventCost.text = Constants.Labels.labelFree
         }
+        _eventCategory.text = event.category
+        
+        cosmosView.settings.fillMode = .half
+        cosmosView.settings.minTouchRating = 0.5
+        cosmosView.rating = 0
+        
+        if event.participants.count <= event.participantNumber {
+            checkUserStatusOnEvent(participants: event.participants)
+        }
+
```

Comment:
This could be separated into multiple methods;

* `setupEventDetails`
* `setupRating`
* `setupEventStatus(for:)`

and then call it inside this `setupUI` method.

No need to change anything here, but just a general suggestion for the future work.